install.sh

```
@@ -33,6 +22,835 @@ print_message "This script will install BirdNET-Go and its dependencies." "$YELL
+    fi
+}
+
+# Function to log messages with timestamps
+log_message() {
+    local level="$1"
+    local message="$2"
+    
+    # Only log if LOG_FILE is set and accessible
+    if [ -n "$LOG_FILE" ] && [ -w "$LOG_FILE" ]; then
+        # Create timestamp in ISO 8601 format
+        local timestamp=$(date '+%Y-%m-%d %H:%M:%S %Z')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)